tribe-ext-map-show-all-events-with-same-venue/src/tribe-events-ajax-maps.js

```diff
@@ -0,0 +1,711 @@
+/**
+ * @file This file contains all map view specific javascript.
+ * This file should load after all vendors and core events javascript.
+ * @version 3.0
+ */
+
+(function( window, document, $, td, te, tf, tg, ts, tt, config, dbug ) {
+
+  /*
+   * $    = jQuery
+   * td   = tribe_ev.data
+   * te   = tribe_ev.events
+   * tf   = tribe_ev.fn
+   * tg   = tribe_ev.geoloc
+   * ts   = tribe_ev.state
+   * tt   = tribe_ev.tests
+   * dbug = tribe_debug
+   */
+  // If we do not have valid geolocation data (ie, the geocenter values are nulls) then let"s
+  // gracefully exit before attempting to interact with the Google Maps API
+  if ( null === TribeEventsPro.geocenter.max_lat ) {
+    return;
+  }
+
+  $.extend( tribe_ev.fn, {
+
+    /**
+     * @function tribe_ev.fn.map_add_marker
+     * @desc tribe_ev.fn.map_add_marker adds event markers to the map on geoloc view.
+     * @param {String} lat Marker latitude.
+     * @param {String} lng Marker longitude.
+     * @param {String} title Marker event title.
+     * @param {String} address Marker event address.
+     * @param {String} link Marker event permalink.
+     */
+
+    map_add_marker: function( lat, lng, title, address, link ) {
+      var myLatlng = new google.maps.LatLng( lat, lng );
+
+      var marker = {
+          position: myLatlng,
+          map     : tg.map
+      };
+
+      // If we have a Map Pin set, we use it
+      if ( "undefined" !== GeoLoc.pin_url && GeoLoc.pin_url ) {
+        marker.icon = GeoLoc.pin_url;
+      }
+
+      // Overwrite with an actual object
+      marker = new google.maps.Marker( marker );
+
+      var infoWindow = new google.maps.InfoWindow();
+
+      var content_title = title;
+
+      var content = content_title;
+
+      if ( address ) {
+        content = content + "<br/>" + "Address: " + address;
+      }
+
+      infoWindow.setContent( content );
+
+      google.maps.event.addListener( marker, "click", function( event ) {
+        infoWindow.open( tg.map, marker );
+      } );
+
+      tg.markers.push( marker );
+
+      if ( tg.refine ) {
+        marker.setVisible( false );
+      }
+      tg.bounds.extend( myLatlng );
+    }
+  } );
+
+
+
+  tg.geocoder = new google.maps.Geocoder();
+  tg.bounds = new google.maps.LatLngBounds();
+
+  $( document ).ready( function() {
+
+
+    /**
+     * @function tribe_test_location
+     * @desc tribe_test_location clears the lat and lng values in event bar if needed. Also hides or shows the geofence filter if present.
+     */
+
+    function tribe_test_location() {
+
+      if ( $( "#tribe-bar-geoloc" ).length ) {
+        var val = $( "#tribe-bar-geoloc" ).val();
+          $fence = $( "#tribe_events_filter_item_geofence" );
```

Comment:
In the original, we have this:

```js
var val = $( '#tribe-bar-geoloc' ).val(),
	$fence = $( "#tribe_events_filter_item_geofence" ),
	$latlng = $( '#tribe-bar-geoloc-lat, #tribe-bar-geoloc-lng' );
```

(One var statement covering multiple variables) but in this version we have multiple separate semi-colon separated var statements ... however, only the first of those uses the `var` command. For scoping reasons we either need to return to the previous form or else add `var` to each line, like this:

```js
var val = $( "#tribe-bar-geoloc" ).val();
var $fence = $( "#tribe_events_filter_item_geofence" );
var $latlng = $( "#tribe-bar-geoloc-lat, #tribe-bar-geoloc-lng" );
```

Same thing applies in a few other places, too.